File: AesCrypter.php
<?php

class AesCrypter {

    private $key = 'php1234567890';
    private $algorithm;
    private $mode;

    public function __construct($key = '', $algorithm = MCRYPT_RIJNDAEL_128,
        $mode = MCRYPT_MODE_CBC) {
        if (!empty($key)) {
            $this->key = $key;
        }
        $this->key = hash('sha256', $this->key, true);
        $this->algorithm = $algorithm;
        $this->mode = $mode;
    }

    public function encrypt($orig_data) {
        $encrypter = mcrypt_module_open($this->algorithm, '',
            $this->mode, '');
        $orig_data = $this->pkcs7padding(
            $orig_data, mcrypt_enc_get_block_size($encrypter)
        );
        mcrypt_generic_init($encrypter, $this->key, substr($this->key, 0, 16));
        $ciphertext = mcrypt_generic($encrypter, $orig_data);
        mcrypt_generic_deinit($encrypter);
        mcrypt_module_close($encrypter);
        return base64_encode($ciphertext);
    }

    public function decrypt($ciphertext) {
        $encrypter = mcrypt_module_open($this->algorithm, '',
            $this->mode, '');
        $ciphertext = base64_decode($ciphertext);
        mcrypt_generic_init($encrypter, $this->key, substr($this->key, 0, 16));
        $orig_data = mdecrypt_generic($encrypter, $ciphertext);
        mcrypt_generic_deinit($encrypter);
        mcrypt_module_close($encrypter);
        return $this->pkcs7unPadding($orig_data);
    }

    public function pkcs7padding($data, $blocksize) {
        $padding = $blocksize - strlen($data) % $blocksize;
        $padding_text = str_repeat(chr($padding), $padding);
        return $data . $padding_text;
    }

    public function pkcs7unPadding($data) {
        $length = strlen($data);
        $unpadding = ord($data[$length - 1]);
        return substr($data, 0, $length - $unpadding);
    }

}

File: AesCrypter.py
# coding: utf-8
import hashlib
from Crypto.Cipher import AES
import base64

class AesCrypter(object):

    def __init__(self, key):
        self.key = hashlib.sha256(key).digest()
        self.iv = self.key[:16]

    def encrypt(self, data):
        data = self.pkcs7padding(data)
        cipher = AES.new(self.key, AES.MODE_CBC, self.iv)
        encrypted = cipher.encrypt(data)
        return base64.b64encode(encrypted)

    def decrypt(self, data):
        data = base64.b64decode(data)
        cipher = AES.new(self.key, AES.MODE_CBC, self.iv)
        decrypted = cipher.decrypt(data)
        decrypted = self.pkcs7unpadding(decrypted)
        return decrypted

    def pkcs7padding(self, data):
        bs = AES.block_size
        padding = bs - len(data) % bs
        padding_text = chr(padding) * padding
        return data + padding_text

    def pkcs7unpadding(self, data):
        lengt = len(data)
        unpadding = ord(data[lengt - 1])
        return data[0:lengt-unpadding]

if __name__ == '__main__':
    aes = AesCrypter('909ed2d5fcf907c79fb9aa341a98febb65291c39')
    print aes.encrypt('AABBCC测试数据')
